op-program/client/l2/engineapi/l2_engine_api.go

```diff
@@ -329,7 +336,24 @@ func (ea *L2EngineAPI) forkchoiceUpdated(ctx context.Context, state *eth.Forkcho
 	// reason.
 	block := ea.backend.GetBlockByHash(state.HeadBlockHash)
 	if block == nil {
-		// TODO: syncing not supported yet
+		if ea.downloader == nil {
+			ea.log.Warn("Must register downloader to be able to snap sync")
+			return STATUS_SYNCING, nil
+		}
+		// If the head hash is unknown (was not given to us in a newPayload request),
+		// we cannot resolve the header, so not much to do. This could be extended in
+		// the future to resolve from the `eth` network, but it's an unexpected case
+		// that should be fixed, not papered over.
```

Comment:
This comment was copied from geth right? It happens to be wrong but geth implements the engine API wrong by requiring `engine_newPayload` before `engine_forkChoiceUpdated`. But given we're mirroring geth's functionality here it's fine.

op-program will never actually snap sync so it only affects the e2e action tests.

Reply:
Yea, this is stripped down from geth. It's probably good that our e2e tests match geth.

I was kinda surprised op-program actually used this code TBH